### PR TITLE
LineSymbolizer GraphicFill and GraphicStroke

### PR DESCRIPTION
**Pull request by jansule**
**Important:** Please wait with merging this PR until
- [x] [this PR from geostyler-style](https://github.com/terrestris/geostyler-style/pull/46)
- [x] [this PR from geostyler-sld-parser](https://github.com/terrestris/geostyler-sld-parser/pull/67)
were merged.

With this PR, geostyler can handle graphicFill and graphicStroke of a LineSymbolizer. This should allow defining styles as described in [this issue](https://github.com/terrestris/geostyler/issues/242). So far, two kinds of styles can be added to graphicFill and graphicStroke: wellKnownSymbol or Icon.

New Components for handling graphicFill and graphicStroke were added:
- **GraphicTypeField:** selecting between supported GraphicTypes (so far `Mark` and `Icon`)
- **GraphicEditor:** handling display of fields depending on GraphicTypeField selection (using `MarkEditor` or `IconEditor`)

Options in `GraphicTypeField` can be deselected. The property will then be removed from the style.

To keep the LineSymbolizer UI compact, antd's `Collapse` component was used to collapse/expand different sections within LineSymbolizer. This is subject of change, with regard to [this comment](https://github.com/terrestris/geostyler/issues/355#issuecomment-414199717).

**Note:** Openlayers API does not natively support graphicFill and graphicStroke. Thus, geostyler-openlayers-parser was not updated and the Preview does not render changes in graphicFill or graphicStroke. However, geostyler-sld-parser does parse graphicStroke and graphicFill.

**Example 1:** Expanded General section, Collapsed GraphicFill and GraphicStroke section
image

**Example 2:** Collapsed General, Expanded GraphicFill and GraphicStroke without anything selected
image

**Example 3:** GraphicFill with WellKnownSymbol, GraphicStroke with Icon
image

**Example 4:** Excerpt from CodeEditor with GraphicFill Icon
image

### CODE REVIEW

File: src/Component/Symbolizer/GraphicEditor/GraphicEditor.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  });
+
+  it('returns CircleSymbolizer as default MarkGraphic', () => {
+    expect(wrapper.instance().getDefaultMarkGraphic()).toEqual(dummyGraphic);
+  });
+
+  it('returns IconSymbolizer as default IconGraphic', () => {
+    const dummyIcon: IconSymbolizer = {
+      kind: 'Icon',
+      image: 'https://upload.wikimedia.org/wikipedia/commons/6/67/OpenLayers_logo.svg'
```
Comment (dnlkoch) [inline src/Component/Symbolizer/GraphicEditor/GraphicEditor.spec.tsx]: Do you think it is possible to pass a local resource here?

File: src/Component/Symbolizer/GraphicEditor/GraphicEditor.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  //   // const onGraphicChangeSpy = jest.spyOn(wrapperInstance.props, 'onGraphicChange');
+  //   wrapperInstance.onGraphicTypeChange('Mark');
+  //   wrapperInstance.onGraphicTypeChange('Icon');
+  //   wrapperInstance.onGraphicTypeChange('Wrong');
+  //   wrapperInstance.onGraphicTypeChange();
+  //   expect(wrapperInstance.props.onGraphicChange).toHaveBeenCalledTimes(4);
+
+  //   // onGraphicChangeSpy.mockReset();
+  //   // onGraphicChangeSpy.mockRestore();
+  // });
```
Comment (dnlkoch) [inline src/Component/Symbolizer/GraphicEditor/GraphicEditor.spec.tsx]: To get this running as expected you may want to pass a `spy` function as prop to the wrapper.

File: src/Component/Symbolizer/GraphicEditor/GraphicEditor.tsx [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+  onGraphicChange: ((graphic: PointSymbolizer) => void);
+}
+
+/** GraphicEditor to select between different graphic options */
+class GraphicEditor extends React.Component <GraphicEditorProps, {}> {
+
+  public static defaultProps: DefaultGraphicEditorProps = {
+    graphicTypeFieldLabel: 'Graphic-Type',
+    iconEditorProps: {
+      defaultIconSource: 'https://upload.wikimedia.org/wikipedia/commons/6/67/OpenLayers_logo.svg'
```
Comment (dnlkoch) [inline src/Component/Symbolizer/GraphicEditor/GraphicEditor.tsx]: I would also prefer a local dummy/fallback icon here.

Comment (jansule) [reply src/Component/Symbolizer/GraphicEditor/GraphicEditor.tsx]: Added local icon and updated path in [this commit](https://github.com/terrestris/geostyler/pull/362/commits/b23d583e15ee6c6e73d1f37f47ba44a7a3126e88). Please note that I also changed the path to the icon in all other files, where a remote source was used.

**Comment by jansule** [comment]
@terrestris/devs ready to merge.

**Comment by KaiVolland** [comment]
Thanks.

**Pull request merged**